reference/reference.go

```diff
@@ -35,7 +35,7 @@ const (
 
 var (
 	// ErrReferenceInvalidFormat represents an error while trying to parse a string as a reference.
-	ErrReferenceInvalidFormat = errors.New("invalid reference format")
+	ErrReferenceInvalidFormat = errors.New("invalid reference format: repository name must be lowercase")
```

Comment:
It's definitely not perfect; I wasn't sure to which extend we wanted to point out the exact reason (mentioning the accepted format would be an alternative).

What should we do if the regex fails and re-running lowercased doesn't make it pass? Do we need a dedicated `ErrReferenceContainsUppercase` ? (because the text is "hardcoded" in the error now)